Creating Mannheimia haemolytica sequencing baits
------------

"The overall goal of this project is to support an accurate, reproducible, transparent and uniform approach to whole-genome sequence (WGS) analysis for purposes of outbreak detection and pathogen surveillance. The overarching objective is to demonstrate how different analytic approaches to whole-genome sequence analysis can impact analysis results. Supporting objectives are to evaluate the impacts of 1) dataset, 2) core- vs. pan-genome inclusion, and 3) genome comparison approach (i.e., using SNPs, k-mers, gene-by-gene alleles, or functional domains)."


Main project repository
-----
[WGS pipeline github](https://github.com/TheNoyesLab/WGS_SNP_pipelines)


Current Tasks
-----

  1. Create pipeline with all WGS analysis tools
  2. Run on Listeria genomes
  
Milestones
-----
![Milestones](https://github.com/EnriqueDoster/project_lab_notebooks/blob/master/postdoc_projects/docs/FMPRE_WGS_timeline.png "timeline")

| Status | Milestones| Date  | Description  |
| -------| ------------- |:------------:| ------------|
| | 1      | Jun 15, 2019  | Download all data, create 15 datasets |
| | 1.5 | ASAP  | Create 15 datasets |
|X| 2      | July 31, 2019 | Download CFSAN-SNP, Lyve-SET, Enterobase, kSNP, Moura et al |
|X| 3      | Aug 1, 2019   | Interim report due |
| | 4      | Sep 30, 2019  | Develop automated pipeline for functional annotations of domains with IBM |
|X| 5      | Nov 30, 2019  | Execute pipelines on the “Random” Salmonella dataset; troubleshooting and validation  |
| | 6      | Jan 31, 2020  | Execute pipelines on all 15 datasets |
| | 7      | May 15, 2020  | Analysis |
| | 8      | Aug 30, 2020  | Manuscript and report preparation |
    
    
    
Project status
-----

- December 2019: 
  * Goal:
  * Notes:
    * 
- November 2019:
  * Goal: Create design by splitting genomes into smaller groups.
  * Notes:
    * 
  * Accomplished: 
    * 
- October 2019:
  * Goal: Figure out way to identify all SRA values for genometrakr
  * Notes:
    * 
  * Accomplished: 
    * 
- September 2019:
  * Goal: Figure out way to identify all SRA values for genometrakr
  * Notes:
    * 
  * Accomplished: 
    * 
- August 2019:
  * Goal: Figure out way to identify all SRA values for genometrakr
  * Notes:
    * 
  * Accomplished: 
    * 
- July 2019:
  * Goal: Figure out way to identify all SRA values for genometrakr
  * Notes:
    * 
  * Accomplished: 
    *    


***
Lab journal
---------------------------------------------------------------------------------------------------------------



### 2019-5-5

* Troubleshooting the download of SRA data and metadata
* Attempted a python script to parse SRA metadata and sent Meera some example files, but still need to troubleshoot duplicate rows.





