Creating Mannheimia haemolytica sequencing baits
------------

"The overall goal of this project is to support an accurate, reproducible, transparent and uniform approach to whole-genome sequence (WGS) analysis for purposes of outbreak detection and pathogen surveillance. The overarching objective is to demonstrate how different analytic approaches to whole-genome sequence analysis can impact analysis results. Supporting objectives are to evaluate the impacts of 1) dataset, 2) core- vs. pan-genome inclusion, and 3) genome comparison approach (i.e., using SNPs, k-mers, gene-by-gene alleles, or functional domains)."


Main project repository
-----
[WGS pipeline github](https://github.com/TheNoyesLab/WGS_SNP_pipelines)


Current Tasks
-----

  1. Create pipeline with all WGS analysis tools
  
  
Milestones
-----
![Miles](https://drive.google.com/drive/u/2/folders/0AFYtL-QXzc3uUk9PVA)

  1. Create pipeline with all WGS analysis tools
    

Current project status
-----

- December 2019: 
  * Goal:
  * Notes:
    * 
- November 2019:
  * Goal: Create design by splitting genomes into smaller groups.
  * Notes:
    * 
  * Accomplished: 
    * 
- October 2019:
  * Goal: Figure out way to identify all SRA values for genometrakr
  * Notes:
    * 
  * Accomplished: 
    * 



***
Lab journal
---------------------------------------------------------------------------------------------------------------


### 2019-8-13

* Probe design resulted in a total of 24,676 baits using 34 M. haemolytica genomes


### 2019-8-12

* Installed CATCH using anaconda
* Run "design.py" on 34 M. haemolytica genomes to test probe creation.

```bash
design.py  all_34_genomes.fa -pl 120 -ps 1 -o test_34genome_probes
```

### 2019-8-9





